core/sigagg/sigagg.go

```diff
@@ -43,47 +41,32 @@ func New(threshold int) *Aggregator {
 // into an aggregated signed duty data object ready to be broadcasted.
 type Aggregator struct {
 	threshold int
-	subs      []func(context.Context, core.Duty, core.PubKey, core.AggSignedData) error
+	subs      []func(context.Context, core.Duty, core.PubKey, core.SignedData) error
 }
 
 // Subscribe registers a callback for aggregated signed duty data.
-func (a *Aggregator) Subscribe(fn func(context.Context, core.Duty, core.PubKey, core.AggSignedData) error) {
+func (a *Aggregator) Subscribe(fn func(context.Context, core.Duty, core.PubKey, core.SignedData) error) {
 	a.subs = append(a.subs, fn)
 }
 
 // Aggregate aggregates the partially signed duty data for the DV.
-func (a *Aggregator) Aggregate(ctx context.Context, duty core.Duty, pubkey core.PubKey, parSigs []core.ParSignedData) error {
-	if len(parSigs) < a.threshold {
+func (a *Aggregator) Aggregate(ctx context.Context, duty core.Duty, pubkey core.PubKey, shares []core.ParSignedData) error {
```

Comment:
`shares` doesn't fit here i think, since we also call partial key shares as `shares`. Maybe rename it to `pSigData` or revert this change